pvt/helpers/src/math/base.ts

```diff
@@ -0,0 +1,158 @@
+import { fp, fpDiv, fpMul } from '../numbers';
+
+export function computeInvariantMock(balances: bigint[]): bigint {
+  // inv = x + y
+  let invariant = 0n;
+  for (let i = 0; i < balances.length; i++) {
+    invariant = invariant + balances[i];
+  }
+
+  return invariant;
+}
+
+export function computeBalanceMock(balances: bigint[], tokenInIndex: number, invariantRatio: bigint): bigint {
+  const invariant = computeInvariantMock(balances);
+  return balances[tokenInIndex] + fpMul(invariant, invariantRatio) - invariant;
+}
+
+export function computeProportionalAmountsIn(
+  balances: bigint[],
+  bptTotalSupply: bigint,
+  bptAmountOut: bigint
+): bigint[] {
+  return balances.map((balance) => (balance * bptAmountOut) / bptTotalSupply);
```

Comment:
I've changed this slightly in #797 introducing `mulDiv`.
The equivalent would be something like this:
```suggestion
  return balances.map((balance) => fpDivUp((balance * bptAmountOut), bptTotalSupply);
```
We don't have rounding up in TS, but perhaps it's worth including it?